File: parrot_spam.rb
#! /usr/bin/env ruby

parrots = ARGV.reject(&:empty?)
if parrots.none?
  STDERR.puts "Nothing to repeat"
  exit 1
end

chars_left = 4000

parrots.cycle do |p|
  chars_left -= p.length
  break if chars_left < 0

  print p
end

puts if STDOUT.isatty

File: parrot-spam.clj
#! /usr/bin/env clj

(def parrots
  (or (not-empty (remove empty? *command-line-args*))
      (do (.println *err* "Nothing to repeat")
          (System/exit 1))))

(defn spam [parrots char-limit]
  (let [all-spam (new StringBuilder)]
    (loop [[p & ps] (cycle parrots)]
      (when (<= (+ (count p) (count all-spam)) char-limit)
        (.append all-spam p)
        (recur ps)))
    (.toString all-spam)))

(println (spam parrots 4000))
